src/components/Api.js

```diff
@@ -0,0 +1,134 @@
+class Api {
+  constructor(options) {
+    // constructor body
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    // if the server returns an error, reject the promise
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+
+  //METHODS FOR GET
+  getInitialCards() {
+    // ...
+    return fetch(`${this._baseUrl}/cards`, {
+        headers: this._headers
+      })
+      .then(res => /*{
+        if (res.ok) {
+          return res.json();
+        }
+        // if the server returns an error, reject the promise
+        return Promise.reject(`Error: ${res.status}`);
+      }*/
+     this._handleResponse(res)
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._handleResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.